ui/src/app/pages/admin/user-audit.tsx

```diff
@@ -38,11 +38,17 @@ const getNextAuditPath = (subject: string) => {
   return `/admin/user-audit/${subject}`;
 };
 
+// Single-user admin page isn't available yet, so go to the main users list page.
+const getAdminPageUrl = (subject: string) => {
```

Comment:
TOL: These "feel" far away from where they are getting used. What are your thoughts on 
1. Since we have the initialSubject and know the query at this time, running the initial queries in these components and passing the data into the AuditPageComponent
2. Changing getAdminPageURL to something like `onAdminClick` or similar - this may be a bit clearer on where and how this property/function is being used

Reply:
for #1, You'd  need a callback  to fetch  new entries on the next subject. How about if we kick that can one PR down the road? I have another batch of changes  right after this. My thought process  was that the user & workspace pages wouldn't need to make any API calls themselves, and would each only contribute what's  unique about their own query. So if we want to  re-fetch  with another subject or another time range, you'd have to call back to the outer component one or two levels up.

Are you saying onAdminClick could have the navigation code directly? I think I like that, assuming I can route from point A to point B from code captured at point C. It's "leaking" knowledge that we're using  the same router in all three places, but that's OK.

Reply:
I am ok kicking  #1 down a bit.

onAdminClick would handle the navigation - it could still accept a parameter. I think it would be clearer at the top level that this is a function being used in a button.